akka/src/main/scala/anorm/AkkaStream.scala

```diff
@@ -72,19 +78,46 @@ object AkkaStream {
    * @param m $materializerParam
    * @param connection $connectionParam
    */
-  def source(sql: => Sql)(implicit m: Materializer, connnection: Connection): Source[Row, NotUsed] = source(sql, RowParser.successful, ColumnAliaser.empty)
+  def source(sql: => Sql)(implicit m: Materializer, connnection: Connection): Source[Row, Future[IOResult]] = source(sql, RowParser.successful, ColumnAliaser.empty)
 
   // Internal stages
 
-  import scala.util.{ Failure, Success }
   import java.sql.ResultSet
-  import akka.stream.{ Attributes, Outlet, SourceShape }
+
   import akka.stream.stage.{ GraphStage, GraphStageLogic, OutHandler }
+  import akka.stream.{ Attributes, Outlet, SourceShape }
+
+  import scala.util.{ Failure, Success }
 
   private[anorm] class ResultSource[T](
       connection: Connection,
       sql: Sql,
       as: ColumnAliaser,
+      parser: RowParser[T],
+      shape: SourceShape[T])(implicit mat: Materializer) extends SourceModule[T, Future[IOResult]](shape) {
```

Comment:
For now I don't see benefit of a new/case specific handling.

Handling the termination, possibly counting the processed element can already be done.

```scala
scala> source.via(Flow.apply.fold(0) { (c, _) => c + 1 }) runWith Sink.last
res1: scala.concurrent.Future[Int]
```